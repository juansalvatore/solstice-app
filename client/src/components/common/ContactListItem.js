import React from 'react'
import styled from 'styled-components'
import Line from './Line'
import PropTypes from 'prop-types'

const ContactListItem = ({ title, data, phoneType }) => (
  <div style={!data ? { display: 'none' } : null}>
    <Line />
    <ListItemWrapper>
      <ItemTitle>{title}</ItemTitle>
      <ListItemDataWrapper>
        <ListItemData>{data}</ListItemData>
        <PhoneType>{phoneType}</PhoneType>
      </ListItemDataWrapper>
    </ListItemWrapper>
  </div>
)

ContactListItem.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.string.isRequired,
  phoneType: PropTypes.string,
}

export default ContactListItem

const ListItemWrapper = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 19px 20px;
  width: 100%;
`
const ItemTitle = styled.span`
  color: rgb(0, 0, 0, 0.3);
  font-weight: 600;
  text-transform: uppercase;
  font-size: 13px;
  margin-bottom: 15px;
`
const ListItemDataWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`
const ListItemData = styled.span``
const PhoneType = styled.span`
  color: rgb(0, 0, 0, 0.5);
  font-weight: 600;
  font-size: 13px;
  display: flex;
  align-items: flex-end;
`
