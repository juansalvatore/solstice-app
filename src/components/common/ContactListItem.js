import React from 'react'
import styled from 'styled-components'
import Line from './Line'

const ContactListItem = ({ title, data, phoneType }) => (
  <div style={!data ? { display: 'none' } : null}>
    <Line />
    <ListItemWrapper>
      <ItemTitle>{title}</ItemTitle>
      <ListItemDataWrapper>
        <ListItemData>{data}</ListItemData>
        <PhoneType>{phoneType}</PhoneType>
      </ListItemDataWrapper>
    </ListItemWrapper>
  </div>
)

export default ContactListItem

const ListItemWrapper = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px 20px;
  width: 100%;
`
const ItemTitle = styled.span`
  color: rgb(0, 0, 0, 0.3);
  font-weight: 600;
  text-transform: uppercase;
  font-size: 13px;
  margin-bottom: 15px;
`
const ListItemDataWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`
const ListItemData = styled.span``
const PhoneType = styled.span`
  color: rgb(0, 0, 0, 0.5);
  font-weight: 600;
  font-size: 13px;
  display: flex;
  align-items: flex-end;
`
