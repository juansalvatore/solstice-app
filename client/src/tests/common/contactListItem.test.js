import React from 'react'
import ContactListItem from '../../components/common/ContactListItem'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('ContactListItem component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<ContactListItem />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
