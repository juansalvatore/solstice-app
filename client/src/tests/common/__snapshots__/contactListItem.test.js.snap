// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactListItem component matches the snapshot 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
>
  <span
    className="sc-bdVaJa ekyfYm"
  />
  <li
    className="sc-bwzfXH hWXwrB"
  >
    <span
      className="sc-htpNat giabQi"
    />
    <div
      className="sc-bxivhb gSSgMV"
    >
      <span
        className="sc-ifAKCX cFlEyZ"
      />
      <span
        className="sc-EHOje TfEqE"
      />
    </div>
  </li>
</div>
`;
